docs/open-api/credentials.yaml

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+---
+
+paths:
+
+  /metalakes/{metalake}/objects/{metadataObjectType}/{metadataObjectFullName}/credentials:
+    parameters:
+      - $ref: "./openapi.yaml#/components/parameters/metalake"
+      - $ref: "./openapi.yaml#/components/parameters/metadataObjectType"
+      - $ref: "./openapi.yaml#/components/parameters/metadataObjectFullName"
+    get:
+      tags:
+        - credentials
+      summary: Get credentials
+      operationId: getCredentials
+      responses:
+        "200":
+          description: Returns the list of credential objects associated with specified metadata object.
+          content:
+            application/vnd.gravitino.v1+json:
+              schema:
+                $ref: "#/components/responses/CredentialResponse"
+              examples:
+                CredentialResponse:
+                  $ref: "#/components/examples/CredentialResponse"
+        "400":
+          $ref: "./openapi.yaml#/components/responses/BadRequestErrorResponse"
+        "5xx":
+          $ref: "./openapi.yaml#/components/responses/ServerErrorResponse"
+
+
+components:
+  schemas:
+    Credential:
+      type: object
+      description: A credential
+      required:
+        - credentialType
+        - expireTimeInMs
+        - credentialInfo
+      properties:
+        credentialType:
```

Comment:
I wouldn't call this `credentialType` because the property has a well-defined scope.
E.g. when I'm modeling a `Person`, I'll simply add a `gender` property rather than
naming it as `personGender`.

This can be pretty simple, e.g.
`{"type": "gcs-token", "expireTimeInMs": 12345, "data": {"token": "secret"}}`

Reply:
Yes, thanks for pointing this out,  I agree with you, how about keeping this as it has been published as API, I will try to use the short style for new features.